src/NuGet.Core/NuGet.Configuration/Settings/Settings.cs

```diff
@@ -545,11 +545,39 @@ private static ISettings LoadSettingsGivenSettingsFiles(List<SettingsFile> setti
                     yield break;
                 }
 
+                // If the default user config NuGet.Config does not exist, we need to create it.
                 var defaultSettingsFilePath = Path.Combine(userSettingsDir, DefaultSettingsFileName);
 
-                // ReadSettings will try to create the default config file if it doesn't exist
+                if (!File.Exists(defaultSettingsFilePath))
+                {
+                    if (!Directory.Exists(userSettingsDir))
+                    {
+                        Directory.CreateDirectory(userSettingsDir);
+                    }
+
+                    File.WriteAllText(defaultSettingsFilePath, NuGetConstants.DefaultConfigContent);
+                    var trackFilePath = Path.Combine(Path.GetDirectoryName(defaultSettingsFilePath), NuGetConstants.V3TrackFile);
+                    File.Create(trackFilePath).Dispose();
+                }
+
                 SettingsFile userSpecificSettings = ReadSettings(rootDirectory, defaultSettingsFilePath, settingsLoadingContext: settingsLoadingContext);
 
+                // Handle when some other product created nuget.config using an old version of NuGet.Core.dll or NuGet.Configuration.dll that did not
+                // automatically add nuget.org as a package source.
+                if (File.Exists(defaultSettingsFilePath) && userSpecificSettings.IsEmpty())
```

Comment:
Better to use an `else if` so we don't check the file we just created unnecessarily?

Reply:
Another advantage of applying Donnie's suggestion is I think also avoid invoking `ReadSettings` method.

Reply:
Writing the default config does not create the in-memory `SettingsFile` object, hence we need to call `ReadSettings` in both code paths. So, either we check the file exists twice, or we call `ReadSettings` in two different places.

Reply:
Why do you need the result again? Aren't you guaranteed the file exists at this point?

Reply:
We need to handle the case that the file already existed, but without any package sources or the tracking file.

Reply:
This won't tell you the file already existed. It will tell you if it exists *now* (after you potentially wrote the file).